File: mobile/library/swift/src/Engine.swift
import Foundation

/// Engine represents a running instance of Envoy Mobile, and provides client interfaces that run on
/// that instance.
@objc
public protocol Engine: AnyObject {
  /// - returns: A client for opening and managing HTTP streams.
  func streamClient() -> StreamClient

  /// A client for recording time series metrics.
  func statsClient() -> StatsClient
}

File: mobile/library/kotlin/src/io/envoyproxy/envoymobile/Engine.kt
package io.envoyproxy.envoymobile

/**
 * Engine represents a running instance of Envoy Mobile, and provides client interfaces that run on
 * that instance.
 */
interface Engine {

  /**
   *  @return a {@link StreamClient} for opening and managing HTTP streams.
   */
  fun streamClient(): StreamClient

  /**
   *  @return a {@link StatsClient} for recording time series metrics.
   */
  fun statsClient(): StatsClient
}
